File: modules/reservation/src/main/java/com/media/bus/reservation/contract/entity/enums/PaymentCycle.java
package com.media.bus.reservation.contract.entity.enums;

import com.media.bus.common.entity.common.BaseEnum;
import lombok.AllArgsConstructor;
import lombok.Getter;

import java.util.Optional;

/// 납부 주기 Enum.
/// - MONTHLY: 월납
/// - QUARTERLY: 분기납
/// - ANNUALLY: 연납
@Getter
@AllArgsConstructor
@SuppressWarnings("unused")
public enum PaymentCycle implements BaseEnum {

    MONTHLY("MONTHLY", "월납"),
    QUARTERLY("QUARTERLY", "분기납"),
    ANNUALLY("ANNUALLY", "연납");

    private final String name;
    private final String displayName;

    public static Optional<PaymentCycle> fromName(String name) {
        return BaseEnum.fromName(PaymentCycle.class, name);
    }
}

File: modules/reservation/src/main/kotlin/com/media/bus/reservation/contract/entity/enums/PaymentCycle.kt
package com.media.bus.reservation.contract.entity.enums

import com.media.bus.common.entity.common.BaseEnum

/**
 * ## 납부 주기 Enum
 *
 * - MONTHLY: 월납
 * - QUARTERLY: 분기납
 * - ANNUALLY: 연납
 */
@Suppress("unused")
enum class PaymentCycle(
    override val displayName: String,
) : BaseEnum {
    MONTHLY("월납"),
    QUARTERLY("분기납"),
    ANNUALLY("연납"),
    ;

    companion object {
        fun fromName(name: String): PaymentCycle? = BaseEnum.fromName<PaymentCycle>(name)
    }
}
